### PR TITLE
Creates Developer Guides section

### PR DESCRIPTION
**Pull request by s-ben**
Created a new top-level menu 'Developer Guides' &mdash;was going to call it 'Blockchain Reference', but this looked better. I'm thinking we can have submenus for several different "Guides", similar to how the Bitcoin dev docs is broken down into [separate guides](https://bitcoin.org/en/developer-guide) (transactions guide, contracts guide, wallets guide, etc.). 

Thinking of creating a few more submenu/Guides as well, but thought I'd run them by you first. Thinking of:

- Developer Guides
  - Transactions
  - Wallets 
  - P2P Network (move this to here?)
  - Mining

### CODE REVIEW

File: docs/developer-guides/addresses.md [hunk 1]
```diff
@@ -36,4 +36,6 @@ The table below provides the locking script and prefix details for each address
 
 ## P2PKH address generation flow
 
+When hashing a public key to generate an address (e.g. a Pay-to-PubkeyHash (P2PKH) address), Decred uses blake256r14 instead of SHA-256. The full algorithm to create an address is blake256r14+ripemd160+base58, as illustated below.
```
Comment (jholdstock) [inline docs/developer-guides/addresses.md]: I think this is a conflict from #35. Can you rebase this PR please.

Comment (s-ben) [reply docs/developer-guides/addresses.md]: Oops, sorry bout that. Rebased

Comment (jholdstock) [reply docs/developer-guides/addresses.md]: A typo has been introduced here, "illustated"

Comment (s-ben) [reply docs/developer-guides/addresses.md]: Fixed

**Comment by s-ben** [comment]
@jholdstock fixed typo. Should be good to go. Thoughts on the additional sub guides I suggested in the comments?

**Pull request merged**